packages/gitgraph-js/src/gitgraph.ts

```diff
@@ -446,164 +544,128 @@ function createGitgraph(
-
-    const tooltip = commit.renderTooltip
-      ? commit.renderTooltip(commit)
-      : createTooltip(commit);
-
-    $tooltip = createG({
-      translate: getWithCommitOffset(commit),
-      children: [tooltip],
-    });
+    function setChildrenForeignObjectHeight(node: Node): void {
+      if (node.nodeName === "foreignObject") {
+        // We have to access the first child's parentElement to retrieve
+        // the Element instead of the Node => we can compute dimensions.
+        const foreignObject = node.firstChild && node.firstChild.parentElement;
+        if (!foreignObject) return;
 
-    svg.firstChild.appendChild($tooltip);
-  }
+        // Force the height of the foreignObject (browser issue)
+        foreignObject.setAttribute(
+            "height",
+            getMessageHeight(foreignObject.firstElementChild).toString(),
+        );
+      }
 
-  function getWithCommitOffset({ x, y }: Coordinate): Coordinate {
-    return { x, y: commitYWithOffsets[y] || y };
+      node.childNodes.forEach(setChildrenForeignObjectHeight);
+    }
   }
 
-  function setBranchLabelRef(commit: Commit, branchLabels: SVGGElement): void {
-    if (!commitsElements[commit.hashAbbrev]) {
-      initCommitElements(commit);
+  protected setBranchLabelRef(commit: Commit, branchLabels: SVGGElement): void {
+    if (!this.commitsElements[commit.hashAbbrev]) {
+      this.initCommitElements(commit);
     }
 
-    commitsElements[commit.hashAbbrev].branchLabel = branchLabels;
+    this.commitsElements[commit.hashAbbrev].branchLabel = branchLabels;
   }
 
-  function setMessageRef(commit: Commit, message: SVGGElement | null): void {
-    if (!commitsElements[commit.hashAbbrev]) {
-      initCommitElements(commit);
+  protected setMessageRef(commit: Commit, message: SVGGElement | null): void {
+    if (!this.commitsElements[commit.hashAbbrev]) {
+      this.initCommitElements(commit);
     }
 
-    commitsElements[commit.hashAbbrev].message = message;
+    this.commitsElements[commit.hashAbbrev].message = message;
   }
 
-  function setTagRef(commit: Commit, tag: SVGGElement): void {
-    if (!commitsElements[commit.hashAbbrev]) {
-      initCommitElements(commit);
+  protected setTagRef(commit: Commit, tag: SVGGElement): void {
+    if (!this.commitsElements[commit.hashAbbrev]) {
+      this.initCommitElements(commit);
     }
 
-    commitsElements[commit.hashAbbrev].tags.push(tag);
+    this.commitsElements[commit.hashAbbrev].tags.push(tag);
   }
 
-  function initCommitElements(commit: Commit): void {
-    commitsElements[commit.hashAbbrev] = {
+  protected initCommitElements(commit: Commit): void {
+    this.commitsElements[commit.hashAbbrev] = {
       branchLabel: null,
       tags: [],
       message: null,
     };
   }
 }
 
+function createGitgraph(
+  graphContainer: HTMLElement,
+  options?: GitgraphOptions,
+  renderer?: Renderer,
+) {
+
+  const gitgraph = new GitgraphCore(options);
+  if (renderer) {
+    renderer.gitgraph = gitgraph;
+  }
```

Comment:
On a different note, but related to previous comment: I'd suggest making `Renderer` an interface that `SvgRenderer` currently implements.

Not 100% sure this is necessary, but at this point it seems that we could accept any kind of `Renderer` implementation that wouldn't necessarily use SVG to render the graph.

Now I said I'm not 100% sure because it adds an abstraction and I'm not sure we need it at this point. The JS library was designed to use SVG and we didn't expected end-users to change that (or they would have created a different rendering library instead). But I can see some value here. Again: I'm not sure because I don't maintain this anymore, so I'll let that decision to you, current users of the lib 👐